api/src/utils/index.js

```diff
@@ -722,83 +722,29 @@ async function notifDepartmentChange(department, template, young, extraParams =
 }
 
 async function addingDayToDate(days, dateStart) {
-  try {
-    const startDate = new Date(dateStart);
-    const newDate = addDays(startDate, days);
-    const formattedValidationDate = newDate.toISOString();
+  const startDate = new Date(dateStart);
+  const newDate = addDays(startDate, days);
+  const formattedValidationDate = newDate.toISOString();
 
-    return formattedValidationDate;
-  } catch (e) {
-    capture(e);
-  }
+  return formattedValidationDate;
 }
 
-async function autoValidationSessionPhase1Young({ young, sessionPhase1, cohort = null, user }) {
-  let cohortWithOldRules = ["2021", "2022", "Février 2023 - C", "Avril 2023 - A", "Avril 2023 - B"];
-  let youngCohort = cohort;
-  if (!cohort) {
-    youngCohort = await CohortModel.findOne({ name: young.cohort });
-  }
-  const {
-    daysToValidate: daysToValidate,
-    validationDate: dateDeValidation,
-    validationDateForTerminaleGrade: dateDeValidationTerminale,
-    dateStart: dateStartcohort,
-  } = await getCohortDateInfo(sessionPhase1.cohort);
+async function autoValidationSessionPhase1Young({ young, sessionPhase1, user }) {
```

Comment:
Je ne maitrise pas ce bout de code, @hlecourt a peut être un avis sur la suppression du legacy.

may be un test pour vérifier le fonctionnement.

sinon pas d'impact sur le endpoint `/young/depart` ?

Reply:
on doit garder malheureusement

Reply:
pour /young/depart je regarde, je comprends pas ton point

Reply:
vu que c'est une méthode commune avec un autre traitement c'est juste que niveau TNR il faut s'assurer que le comportement métier de l'autre fonctionnalité n'est pas impacté. mais du coup si on garde de le code legacy pas de soucis